### ISSUE TITLE
npa-tool does not work in fedora environment

### ISSUE THREAD

**Issue by usual-user** [OpenSC/OpenSC#1630]
"OPENSC_DRIVER=npa npa-tool -vvv" yields [npa-tool-vvv.log](https://github.com/OpenSC/OpenSC/files/2954628/npa-tool-vvv.log).
"OPENSC_DRIVER=npa opensc-tool -vvv --atr" yields a similar result.
"pcsc_scan" yields [pcsc_scan.log](https://github.com/OpenSC/OpenSC/files/2954630/pcsc_scan.log).

As I do not know the propper use of this tools till now it is maybe an user fault.

**Comment by Jakuje** [comment]
The `npa-tool` requires OpenPACE, which is not available in Fedora. I do not have any card to test with, but if you have, I would be happy to build OpenSC with OpenPACE in Fedora.

**Comment by usual-user** [comment]
My fedora has already OpenPACE support, see "[openpace does not work in fedora environment](https://github.com/frankmorgner/openpace/issues/39)" for reference.

**Comment by frankmorgner** [comment]
The typical use in OpenSC is to make private keys usable, which have an X.509 certificate attached. The German ID card has this only after initialization of the eSign application (your card isn't initialized). Unfortunately, there is no certificate service provider available anymore, which offers does this kind of initialization. D-Trust was the last provider, which converted its [sign-me](https://www.bundesdruckerei.de/de/2837-sign-me) into a remote signature service rather than using the ID card for qualified signatures.

I've pushed a change to `npa-tool`, to force the default driver so that all of its functionality can be used independent from the internal `npa` card driver (d953998aa314b648b6f0785f7e062ae80067c05b). 

What do you want to do with the ID card? There are still some things that could be done with the card's standard eID application, for example https://github.com/frankmorgner/eid-pam or https://gitlab.informatik.hu-berlin.de/michalof/eidsacp.

**Comment by frankmorgner** [comment]
Of course, there are also the official use cases for eID at some service providers. The links I've posted above are available for everyone.

**Comment by usual-user** [comment]
The better question would be "What can I do with the German ID card?". As the eID function nowadays is always enabled, I tought "Hey, let's see what use this card is in my environment". My environment is a current fedora LxQT desktop on an arm imx6 platform. So all Windows, Mac, Android or iOS tools are out of scope. Searching the fedora package repository reveal some traces of npa-tool which man page claims "displays information on the German eID card (neuer Personalausweis, nPA)". As I am a total NOOP with opensc let see if I can make it work to display at least some basic information about an attached card. It get identified as "possibly German passport (passport)" by pcsc_scan.
I rebuild opensc with the latest git master, but the messages I get seem still to indicate it is not properly working (sc_connect_card: card info name:'Unsupported card', type:-1, flags:0x0, max_send/recv_size:255/256).
Thank you for pointing mit at eid-pam. It let me revisit AusweisApp2 which lead to https://github.com/Governikus/AusweisApp2.
I composed a SPEC file to build for fedora inspired by AusweisApp2-1.16.1-1.8.src.rpm that google pointed me at. All dependencies are satisfied by fedora out of the box.
This are some sniplets from the resulting log while playing with reader and card attach:
> ApplicationName: "AusweisApp2"
> ApplicationVersion: "1.16.1"
> OrganizationName: "AusweisApp2_CE"
> OrganizationDomain: ""
> System: "Fedora 30 (Thirty)"
> Kernel: "5.0.0-0.rc1.git0.1.fc26.armv7hl"
> Architecture: "arm"
> Devicename: "tinker-box"
> Qt Version: 5.12.1
> OpenSSL Version: "OpenSSL 1.1.1b FIPS  26 Feb 2019"
>
> retrieved retry counter: 3 , was: -1 , PIN deactivated: false
> Card inserted: {Type: EID_CARD, Retry counter: 3, Pin deactivated: false}
> Card removed
> retrieved retry counter: 3 , was: -1 , PIN deactivated: false
> Card inserted: {Type: EID_CARD, Retry counter: 3, Pin deactivated: false}
> Card removed
> SCardGetStatusChange: "Scard_E_Unknown_Reader"
> Reader unknown, stop updating reader information
> Cannot update readers

So my equipment seems to be properly working and status information can be retrieved.

**Comment by frankmorgner** [comment]
Great! I see that 0.16.0 of the AusweisApp was finally released. You should use the libwebseckets branch of eid-pam, which allows remote controlling the user input (PIN). I see that starting AusweisApp in the correct mode has changed a bit. You need to check the userguide on websockets in https://www.ausweisapp.bund.de/fileadmin/user_upload/Software/1.16.0/AusweisApp2-1.16.0-SDK.pdf#subsection.3.1 until I've updated the documenation of eid-pam.

Unfortunately, eID cannot freely be used by normal users due to the authentication restrictions. eid-pam uses the data from the "Selbstauskunft" to authenticate the user. You could [look at it as a bug](http://sar.informatik.hu-berlin.de/research/publications/SAR-PR-2012-05/SAR-PR-2012-05_.pdf) (page 43pp), but in eid-pam we use it as a feature.
